### ISSUE TITLE
Fix S5034 FP: incorrect "ValueTask should be consumed correctly" in Task.WhenAll with different methods on same class instance

### ISSUE THREAD

**Issue by Basssiiie** [SonarSource/sonar-dotnet#6779]
### Description

Rule [S5034](https://rules.sonarsource.com/csharp/RSPEC-5034) reports a false positive when consuming two different ValueTasks from different methods on the same class, within a `Task.WhenAll()` and with a `AsTask()`.

### Repro steps

```cs
class ExampleService
{
	public ValueTask DoThing() => ValueTask.CompletedTask;
	public ValueTask DoAnotherThing() => ValueTask.CompletedTask;

}

class Program
{
	public async Task Main()
	{
		var service = new ExampleService();

		await Task.WhenAll(
			service.DoThing().AsTask(), // <- S5034 error here, even though both lines use different methods.
			service.DoAnotherThing().AsTask()
		);
	}
}
```

### Expected behavior

No warnings here.

### Actual behavior

SonarQube/SonarCloud reports the aforementioned warning on the first line within the `Task.WhenAll()`, even though both `ValueTask`'s are different as they are returned from different methods.

### Related information

* SonarLint Visual Studio plugin version: 6.12.0.59751
* Visual Studio version: 17.4.4
* dotnet version: 7.0.102
* Operating System: Windows 10

Hope it helps. Let me know if there are any further questions. :)

**Comment by gregory-paidis-sonarsource** [comment]
Thank you for opening this issue @Basssiiie .

I can confirm that this is indeed a False Positive. 
I added a [reproducer](https://github.com/SonarSource/sonar-dotnet/pull/6826) in our testcases and included this issue to our backlog.

Kind Regards, Greg.

**Comment by Basssiiie** [comment]
Hello @gregory-paidis-sonarsource 

Thank you. Do you have an idea of where this is on the backlog? It's about a year and a half later and we still have this false positive occurring in our code base.

Thanks! 🙂

**Comment by gregory-paidis-sonarsource** [comment]
Hey there again!
Unfortunately, I cannot give you an estimate.
When we do our next FP/FN hardening sprint, I will try to prioritize this.

**Comment by Basssiiie** [comment]
@gregory-paidis-sonarsource Thanks! 🙂

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1847](https://sonarsource.atlassian.net/browse/NET-1847)

[NET-1847]: https://sonarsource.atlassian.net/browse/NET-1847?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
